src/writeData/subscriptions/context/subscription.update.tsx

```diff
@@ -144,6 +153,27 @@ export const SubscriptionUpdateProvider: FC = ({children}) => {
     update(currentSubscription)
   }
 
+  const setStatus = (isActive: boolean): any => {
+    setLoading(RemoteDataState.Loading)
+    const params = {id: currentID, data: {isActive}}
+    updateStatusAPI(params)
+      .then(() => {
+        getSubscription()
+      })
+      .catch(() => {
+        dispatch(notify(subscriptionStatusUpdateFail()))
```

Comment:
I think you need to `setLoading(RemoteDataState.Done)` here so the spinner doesn't live forever if there is an error

Reply:
good point, i think there's other places where i was doing that too, so fixed in latest commit